apps/mobile/src/features/swap/components/currency-mode-switcher.tsx

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+import Animated, { FadeIn, FadeOut } from 'react-native-reanimated';
+
+import { formatCurrency } from '@/utils/currency-formatter';
+
+import {
+  ArrowTopBottomIcon,
+  Box,
+  Pressable,
+  SkeletonLoader,
+  Text,
+  legacyTouchablePressEffect,
+} from '@leather.io/ui/native';
+import { assertUnreachable } from '@leather.io/utils';
+
+import { SecondaryAmount } from '../swap-state/swap-state.types';
+
+interface CurrencySwitchProps {
+  secondaryAmount: SecondaryAmount;
+  onModeSwitch: () => void;
+}
+
+const AnimatedBox = Animated.createAnimatedComponent(Box);
+
+export function CurrencyModeSwitcher({ secondaryAmount, onModeSwitch }: CurrencySwitchProps) {
+  function renderContent() {
+    switch (secondaryAmount.status) {
+      case 'idle':
+        return <Box width={56} height={20} aria-hidden />;
+      case 'pending':
+        return <PendingIndicator />;
+      case 'error':
+        // TODO: Extremely rare, needs better design.
+        return (
+          <Text variant="label02" color="ink.text-subdued">
+            –
```

Comment:
Maybe we can render the `EmptyBalanceDisplay` from balance here? `-.--`

Reply:
That would nicer, 100%. I want to add a little hint there as well for when market data fails to load, though it is pretty rare. I'll add this to next PR which is all about amounts.